test/unittests/execute_floating_point_test.cpp

```diff
@@ -42,6 +42,134 @@ MATCHER_P(ArithmeticNaN, value, "result with an arithmetic NaN")
+
+    static constexpr auto num_nans = 7;
+    static constexpr auto num_positive = m_values.size() - num_nans;
+    static constexpr auto num_ordered = 2 * num_positive - 1;
+
+    class Range
+    {
+        Iterator m_begin;
+        Iterator m_end;
+
+    public:
+        constexpr Range(Iterator begin, Iterator end) noexcept : m_begin{begin}, m_end{end} {}
+
+        constexpr Iterator begin() const { return m_begin; }
+        constexpr Iterator end() const { return m_end; }
+        constexpr size_t size() const { return static_cast<size_t>(m_end - m_begin); }
+    };
+
+    // The list of positive floating-point values without zero, infinity and NaNs.
+    static Range positive_nonzero_finite() noexcept
+    {
+        return {&m_values[1], &m_values[num_positive - 1]};
+    }
+
+    // The list of positive floating-point values without zero and NaNs (includes infinity).
+    static Range positive_nonzero_infinite() noexcept
+    {
+        return {&m_values[1], &m_values[num_positive]};
+    }
+
+    // The list of positive NaN values.
+    static Range positive_nans() noexcept { return {m_values.end() - num_nans, m_values.end()}; }
+
+    // The list of positive non-canonical NaN values (including signaling NaNs).
+    static Range positive_noncanonical_nans() noexcept
+    {
+        return {m_values.end() - num_nans + 1, m_values.end()};
+    }
+
+    // The list of positive floating-point values with zero, infinity and NaNs.
+    static Range positive_all() noexcept { return {m_values.begin(), m_values.end()}; }
+
+    // The list of floating-point values, including infinities.
+    // They are strictly ordered (ordered_values[i] < ordered_values[j] for i<j).
+    // Therefore -0 is omitted. This allows determining the relation of any pair of values only
+    // knowing values' position in the array.
+    static auto& ordered() noexcept
+    {
+        static const auto ordered_values = [] {
+            std::array<T, num_ordered> a;
+            const auto ps = positive_nonzero_infinite();
+
+            auto it = std::begin(a);
+            it = std::transform(std::reverse_iterator{std::end(ps)},
+                std::reverse_iterator{std::begin(ps)}, it, std::negate<T>{});
+            *it++ = T{0.0};
+            std::copy(std::begin(ps), std::end(ps), it);
+            return a;
+        }();
+
+        return ordered_values;
+    }
+
+    // The list of floating-point values, including infinities and NaNs.
+    // They are strictly ordered (ordered_values[i] < ordered_values[j] for i<j) or NaNs.
+    // Therefore -0 is omitted. This allows determining the relation of any pair of values only
+    // knowing values' position in the array.
+    static auto& ordered_and_nans() noexcept
+    {
+        static const auto ordered_values = [] {
+            const auto& without_nans = ordered();
+            const auto nans = positive_nans();
+
+            std::array<T, num_ordered + 2 * num_nans> a;
```

Comment:
And calculate the size here as `without_nans.size() + 2 * nans.size()`

Reply:
Done differently.